lib/vrl/stdlib/src/encode_json.rs

```diff
@@ -52,8 +52,8 @@ impl Expression for EncodeJsonFn {
         }
     }
 
-    fn type_def(&self, state: &state::Compiler) -> TypeDef {
-        self.value.type_def(state).infallible().bytes()
+    fn type_def(&self, _: &state::Compiler) -> TypeDef {
+        TypeDef::bytes().infallible()
```

Comment:
A `TypeDef` consists of two data points: fallibility, and the `kind`. There were several cases where his was probably unknown to function implementors, and thus you see something like here, where we first get the type def of the `value` variable, and then overwrite all of its data points by using `infallible()` and `bytes()`.

I've rectified these to make it clear what's going on here, but there's no change in behaviour.